BTModules/Office365.psm1

```diff
@@ -0,0 +1,244 @@
+        DisplayCompletedOperation "OK" Green
+    }
+    catch
+    {
+        DisplayCompletedOperation "FAIL" Red
+        Write-Host -ForegroundColor Red "ERROR: Import-PSSession failed with $_"
+        return $false
+    }
+    
+    return $session    
+}
+
+################################################################################
+# Get available SKUs
+################################################################################
+function Get-AvailableAccountSku()
+{
+    try
+    {
+        DisplayInProgressOperation "Get All Office 365 Users Account Sku"
+        $SKUs = Get-MsolAccountSku | Where-Object {$_.ActiveUnits -gt $_.ConsumedUnits}
+        DisplayCompletedOperation "OK" Green
+
+        if ($null -eq $SKUs)
+        {
+            Write-Host -ForegroundColor Red "FATAL: No SKUs available"
+            return $false
+        }
+        
+        return $SKUs
+    }
+    catch
+    {
+        DisplayCompletedOperation "FAIL" Red
+        Write-Host -ForegroundColor Red "ERROR: Get-MsolUser failed with $_"
+        return $false
+    }
+}
+
+################################################################################
+# Get all users from Office 365
+################################################################################
+function Get-Office365Users
+{
+    try
+    {
+        DisplayInProgressOperation "Get All Office 365 Users"
+        $users = Get-MsolUser
+        DisplayCompletedOperation "OK" Green
+        return $users
+    }
+    catch
+    {
+        DisplayCompletedOperation "FAIL" Red
+        Write-Host -ForegroundColor Red "ERROR: Get-MsolUser failed with $_"
+        return $false
+    }
+}
+
+################################################################################
+# Remove a user from Office 365
+################################################################################
+function Remove-Office365User($user)
+{
+    try
+    {
+        DisplayInProgressOperation "Removing $($user.UserPrincipalName)"
+        Remove-MsolUser -UserPrincipalName $user.UserPrincipalName -Force
+        DisplayCompletedOperation "OK" Green
+        return $true
+    }
+    catch
+    {
+        DisplayCompletedOperation "FAIL" Red
+        Write-Host -ForegroundColor Red "ERROR: Remove-MsolUser failed with $_"
+        return $false
+    }
+}
+
+############################################################
+# Create a new user in Office 365
+############################################################
+function New-Office365User($email, $firstName, $lastName, $sku, $defaultPassword)
+{
+    try
+    {
+        $displayName = "$firstName $lastName"
+        DisplayInProgressOperation "Creating $email"
+        New-MsolUser -DisplayName $displayName -FirstName $firstName -LastName $lastName -UserPrincipalName $email -UsageLocation "US" -LicenseAssignment $sku -Password $defaultPassword -ForceChangePassword $false | Out-Null
```

Comment:
Everything is passed in to New-MsolUser except -UsageLocation being statically set to"US".